### PR TITLE
feat(esl-media): add extended `lazy` attribute support⚠️

### PR DESCRIPTION
**Pull request by Natalie-Smirnova**
Closes:  #1641

### CODE REVIEW

File: src/modules/esl-media/README.md [hunk 1]
```diff
@@ -45,8 +45,16 @@ using a single tag as well as work with external providers using simple native-l
  - `ready-class` - class to add when the resource is ready
  - `ready-class-target` - [ESLTraversingQuery](../esl-traversing-query/README.md) to define a target for `ready-class`
 
- - `disabled` (boolean) - marker that prevents media api initialization
- 
+ - `disabled` (boolean) - marker that prevents media api initialization. Deprecated alias for *manual* mode of `lazy` attribute
```
Comment (ala-n) [inline src/modules/esl-media/README.md:48]: TODO: please check indents and correct rendering on the [GitHub](https://github.com/exadel-inc/esl/blob/esl-media-support-lazy-initialization/src/modules/esl-media/README.md) and first of all on the related site page

File: src/modules/esl-media/core/esl-media.ts [hunk 2]
```diff
@@ -159,6 +163,15 @@ export class ESLMedia extends ESLBaseElement {
...
         this.deferredReinitialize();
         break;
+      case 'lazy':
+        if (this.lazy === 'manual') {
+          this.deferredReinitialize();
+        } else if (this.lazy === 'auto') {
+          this.attachViewportConstraint();
+        } else {
+          this.detachViewportConstraint();
+        }
```
Comment (dshovchko) [inline src/modules/esl-media/core/esl-media.ts:173]: In order to reduce the complexity of the method, can we put this code in a separate method, for example, `onLazyAttributeChange()`?

File: src/modules/esl-media/core/esl-media-iobserver.ts [hunk 3]
```diff
@@ -11,21 +13,25 @@ export function getIObserver(lazy: boolean = false): IntersectionObserver {
...
-      threshold: [RATIO_TO_DEACTIVATE, RATIO_TO_ACTIVATE]
+      threshold: [RATIO_TO_STOP, RATIO_TO_PLAY]
     });
   }
   return iObserver;
 }
 
 function handleViewport(entry: IntersectionObserverEntry): void {
   const video = entry.target as ESLMedia;
+  // Remove `lazy` attribute when media is in viewport with min ratio RATIO_TO_ACTIVATE
```
Comment (dshovchko) [inline src/modules/esl-media/core/esl-media-iobserver.ts:24]:
```suggestion
  // Removes `lazy` attribute when media is in the viewport with min ratio RATIO_TO_ACTIVATE
```

**Review by dshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by ala-n — CHANGES_REQUESTED** [review]
Please check the following points:
 - `<esl-media lazy ...` or `<esl-media lazy="blabla">` shod be considered as lazy = 'auto'
 - Please add at least one example to check lazy, for example make the background video examples on the embedded-video page lazy
 - As for now, the lazy loading seems to be not working the video on the page does not load at the moment of lazy attribute removal

**Review by ala-n — CHANGES_REQUESTED** [review]
⛔ live tests are not passed: the `esl-media` instances are not lazy.

**Review by ala-n — APPROVED** [review]
✅ Live Tests are passed no additional requests from my side

**Review by dshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-media/test/esl-media.lazy.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    await promisifyTimeout(100);
+    $media.querySelectorAll('video').forEach((v) => v.dispatchEvent(new Event('loadedmetadata')));
+    await promisifyTimeout(0);
+
+    expect($media.ready).toBe(true);
+    expect($media.querySelectorAll('*')).toEqual(expect.objectContaining({length: 1}));
+  });
+
+  test('ESLMedia is not loading when lazy atrribute is set', async () => {
```
Comment (abarmina) [inline src/modules/esl-media/test/esl-media.lazy.test.ts:28]:
```suggestion
  test('ESLMedia is not loading when lazy attribute is set to manual', async () => {
```

**Review by abarmina — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Pull request merged**